src/main/java/uk/ac/ebi/subs/processing/dispatcher/DispatcherServiceImpl.java

```diff
@@ -85,7 +85,7 @@
                 SubmittableRepository submittableRepository = dataTypeRepositoryMap.get(type);
 
                 for (String submittableId : typeAndIds.getValue()) {
-                    final Optional<?> submittableById = submittableRepository.findById(submittableId);
+                    final Optional<?> submittableById = Optional.of(submittableRepository.findOne(submittableId));
```

Comment:
please use Optional.ofNullable(). otherwise the api throws an error if the passed argument is null. i experienced this myself.